### ISSUE TITLE
Trilinos nightly test failures in Stokhos unit tests

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos-kernels#1631]
Nightly Trilinos builds began failing in several Stokhos unit tests following merge of #1618 :

```
   44 - Stokhos_KokkosCrsMatrixUQPCEUnitTest_Cuda_MPI_1 (Failed)
   45 - Stokhos_TpetraCrsMatrixUQPCEUnitTest_Cuda_MPI_4 (Failed)
   52 - Stokhos_KokkosCrsMatrixMPVectorUnitTest_Cuda_MPI_1 (Failed)
```

Rebuilding with PR #1628 did not resolve the failures for me.

**Reproducer** (weaver rhel8 queue):
Trilinos: use kokkos-promotion branch
(Some of the module load and package enables may be extraneous)
```bash
# Repos
git clone -b kokkos-promotion https://github.com/trilinos/Trilinos.git
git clone -b develop https://github.com/kokkos/kokkos.git
git clone -b develop https://github.com/kokkos/kokkos-kernels.git
# Symbolic link to your kokkos and kokkos-kernels repos in Trilinos source directory for source override
cd Trilinos
ln -s <path-to-your-repo>/kokkos kokkos
ln -s <path-to-your-repo>/kokkos-kernels kokkos-kernels

cd $HOME
mkdir -p build
cd build

module load git cmake/3.21.2 cuda/11.2.2/gcc/8.3.1 openmpi/4.1.1/gcc/8.3.1/cuda/11.2.2 openblas/0.3.18/gcc/8.3.1 boost/1.70.0/gcc/8.3.1
export OMPI_CXX=${KOKKOS_PATH}/bin/nvcc_wrapper

TESTS=OFF

cmake \
 -D CMAKE_CXX_COMPILER=`which mpicxx` \
 -D CMAKE_CXX_FLAGS="-g" \
 -D CMAKE_C_COMPILER=`which mpicc` \
 -D CMAKE_C_FLAGS="-g" \
 -D CMAKE_Fortran_COMPILER=`which mpifort` \
 -D CMAKE_Fortran_FLAGS="-g" \
 -D CMAKE_BUILD_TYPE:STRING=RELEASE \
 -D CMAKE_CXX_STANDARD=17 \
 -D BUILD_SHARED_LIBS=OFF \
\
 -D TPL_ENABLE_CUDA:STRING=ON \
 -D TPL_ENABLE_MPI:STRING=ON \
 -D MPI_BASE_DIR:PATH="$OPENMPI_ROOT" \
 -D MPI_BIN_DIR:PATH="$OPENMPI_BIN" \
 -D MPI_EXEC_POST_NUMPROCS_FLAGS:STRING="-map-by;socket:PE=4" \
\
-D TPL_ENABLE_BLAS:STRING=ON \
  -D BLAS_LIBRARY_DIRS:FILEPATH="${OPENBLAS_ROOT}/lib" \
  -D BLAS_LIBRARY_NAMES:STRING="openblas" \
-D TPL_ENABLE_LAPACK:STRING=ON \
  -D LAPACK_INCLUDE_DIRS:FILEPATH="${OPENBLAS_ROOT}/include" \
  -D LAPACK_LIBRARY_DIRS:FILEPATH="${OPENBLAS_ROOT}/lib" \
  -D LAPACK_LIBRARY_NAMES:STRING="openblas" \
-D TPL_ENABLE_Boost=ON \
   -D Boost_INCLUDE_DIRS:PATH="${BOOST_ROOT}/include" \
   -D Boost_LIBRARY_DIRS:PATH="${BOOST_ROOT}/lib" \
-D TPL_ENABLE_BoostLib=ON \
   -D BoostLib_INCLUDE_DIRS:PATH="${BOOST_ROOT}/include" \
   -D BoostLib_LIBRARY_DIRS:PATH="${BOOST_ROOT}/lib" \
-D TPL_ENABLE_Netcdf=OFF \
\
 -D Trilinos_ENABLE_COMPLEX=ON \
 -D Trilinos_ENABLE_TESTS=OFF \
 -D Trilinos_ENABLE_EXAMPLES=OFF \
\
 -D Trilinos_ENABLE_Kokkos=ON \
 -D Kokkos_ENABLE_CUDA=ON \
 -D Kokkos_ENABLE_CUDA_LAMBDA=ON \
 -D Kokkos_ENABLE_CUDA_UVM=ON \
 -D Kokkos_ARCH_VOLTA70=ON \
 -D Kokkos_ARCH_POWER9=ON \
 -D Kokkos_ENABLE_TESTS=${TESTS} \
 -D Trilinos_ENABLE_KokkosKernels=ON \
 -D KokkosKernels_ENABLE_TESTS=${TESTS} \
 -D Trilinos_ENABLE_Tpetra=ON \
 -D Tpetra_ENABLE_CUDA=ON \
 -D Tpetra_ENABLE_SERIAL=ON \
 -D Tpetra_ENABLE_TESTS=${TESTS} \
 -D Trilinos_ENABLE_Sacado=ON \
 -D Sacado_ENABLE_TESTS=${TESTS} \
 -D Trilinos_ENABLE_Stokhos=ON \
 -D Stokhos_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Belos=ON \
 -D Belos_ENABLE_TESTS=${TESTS} \
\
 -D Trilinos_ENABLE_Ifpack2=ON \
 -D Ifpack2_ENABLE_TESTS=${TESTS} \
 -D Trilinos_ENABLE_Amesos2=ON \
 -D Amesos2_ENABLE_TESTS=${TESTS} \
 -D Trilinos_ENABLE_Anasazi=ON \
 -D Trilinos_ENABLE_Teuchos=ON \
 -D Teuchos_ENABLE_TESTS=${TESTS} \
 -D Trilinos_ENABLE_Intrepid2=ON \
 -D Intrepid2_ENABLE_TESTS=${TESTS} \
 -D Trilinos_ENABLE_MueLu=ON \
 -D MueLu_ENABLE_TESTS=${TESTS} \
 -D Trilinos_ENABLE_Panzer=ON \
 -D Panzer_ENABLE_TESTS=${TESTS} \
\
 -D Trilinos_ENABLE_ShyLU_NodeTacho=OFF \
 -D Trilinos_ENABLE_Zoltan2Sphynx=OFF \
 -D Trilinos_ENABLE_SEACAS=OFF \
\
  -DKokkos_SOURCE_DIR_OVERRIDE:STRING=kokkos \
  -DKokkosKernels_SOURCE_DIR_OVERRIDE:STRING=kokkos-kernels \
  -DKokkosKernels_ENABLE_SPARSE=ON \
  -DKokkosKernels_ENABLE_GRAPH=ON \
  -DKokkosKernels_ENABLE_BLAS=ON \
  -DKokkosKernels_ENABLE_BATCHED=ON \
$TRILINOS_DIR
```

**Comment by brian-kelley** [comment]
@ndellingwood Thanks! I have a build going now on weaver. Just a heads up, I had to comment out some CMake logic that just went into kokkos/CMakeLists.txt a couple days ago, from https://github.com/kokkos/kokkos/pull/5463:
```
199 # Gather information about the runtime environment
200 INCLUDE(${KOKKOS_SRC_PATH}/cmake/build_env_info.cmake)
201 #check_git_setup() #this line errored out with stuff below:
```

```
-- Configuring done
CMake Error: install(EXPORT "Kokkos" ...) includes target "Kokkos_all_libs" which requires target "impl_git_version" that is not in any export set.
CMake Error in packages/kokkos/CMakeLists.txt:
  export called with target "Kokkos_all_libs" which requires target
  "impl_git_version" that is not in any export set.
```

**Comment by ndellingwood** [comment]
@brian-kelley thanks for the warning! That just started popping up in the nightlies as well ( https://jenkins-son.sandia.gov/view/Kokkos/job/Kokkos_Trilinos_develop_kokkosdev2_cuda_111_opt/803/ ), I'll file an issue so it can be addressed before 4.0

**Comment by ndellingwood** [comment]
Cross-reference #1633
